src/parser/Parser.ts

```diff
@@ -1316,7 +1316,7 @@ export class Parser {
         function prefixUnary(): Expression {
             if (match(Lexeme.Not, Lexeme.Minus)) {
                 let operator = previous();
-                let right = prefixUnary();
+                let right = relational();
```

Comment:
Considering the order of operations goes `relational` > `bitshift` > `additive` > `multiplicative` > `exponential` > `prefixUnary`, I was going to add some unit tests that did things like `not 2 * 3`. However, that elicits strange behavior on RBI (`not 2 * 3 = -7` for some reason...) and `brs` prevents comparing boolean `NOT` to non-boolean `2 * 3`.

Reply:
🤔  interesting!

Reply:
Hmm, seems like there's still some bugs in `brs` that we'll need to shake out then! From RBI:

```brightscript
print not  0 ' => -1
print not  1 ' => -2
print not -1 ' =>  0
print not  2 ' => -3
print not -2 ' =>  1
```

which looks like RBI is using a [two's complement](https://en.wikipedia.org/wiki/Signed_number_representations#Two's_complement) representation for numbers and does something slightly unexpected with it.  `not someInteger` in RBI seems to take `someInteger`, convert to a binary representation, perform a bitwise negation on it (all 0s become 1s, all 1s become 0s), then convert back to a decimal representation in the Two's complement understanding.  So for `not 0` (using just 4-bits for now because I'm lazy), we have:

```
not 0 (decimal) == not 0b0000 == 0b1111 == -1 (decimal in two's complement)
```

How strange!  In the `not 2 * 3` example, we get:

```
not (2 * 3) == not 6 (decimal) == not 0b0110 == 0b1001 == -7 (decimal in two's complement)
```

All-up, it looks like RBI supports negating decimals, but `brs` doesn't!  I'll file a bug to fix that a bit later.  At least the parser is still correct 🎉